resource-aggregate/service/aggregate.go

```diff
@@ -17,25 +17,27 @@ type LogPublishErrFunc func(err error)
 
 type Aggregate struct {
 	ag         *cqrsAggregate.Aggregate
+	hubID      string
 	eventstore EventStore
 }
 
 func ResourceStateFactoryModel(context.Context) (cqrsAggregate.AggregateModel, error) {
-	return raEvents.NewResourceStateSnapshotTaken(), nil
+	return events.NewResourceStateSnapshotTaken(), nil
 }
 
 func ResourceLinksFactoryModel(context.Context) (cqrsAggregate.AggregateModel, error) {
-	return raEvents.NewResourceLinksSnapshotTaken(), nil
+	return events.NewResourceLinksSnapshotTaken(), nil
 }
 
 func DeviceMetadataFactoryModel(context.Context) (cqrsAggregate.AggregateModel, error) {
-	return raEvents.NewDeviceMetadataSnapshotTaken(), nil
+	return events.NewDeviceMetadataSnapshotTaken(), nil
 }
 
 // NewAggregate creates new resource aggreate - it must be created for every run command.
-func NewAggregate(resourceID *commands.ResourceId, snapshotThreshold int, eventstore EventStore, factoryModel cqrsAggregate.FactoryModelFunc, retry cqrsAggregate.RetryFunc) (*Aggregate, error) {
+func NewAggregate(resourceID *commands.ResourceId, snapshotThreshold int, hubID string, eventstore EventStore, factoryModel cqrsAggregate.FactoryModelFunc, retry cqrsAggregate.RetryFunc) (*Aggregate, error) {
```

Comment:
When creating an aggregate, you always pass the hubID from the configuration. When you then load events, you do not care if it is the matching one or not. And when handling command and publishing the event, you always take the hub id from the context -> aggregate -> configuration, while in identity store you take the one stored into the db. You use two different approaches. Please come up with a consistent solution - e.g. updating the identity store.

Reply:
Fixed -> removed from IS storage